File: Assets/Scripts/Spells/Straight.js
#pragma strict
var speed : float = 4;
var moving : boolean = false;
var finished : boolean = false;
var target : Vector3;

private var incr : float = 0;

function Update () {
	if (finished)
		Destroy(gameObject, .01);
	
	if (moving)
	{
		if(incr <=1)
		    incr += speed/100;
		else
			finished = true;
		    
		transform.position = Vector3.Lerp(transform.position, target, incr);
	}

}

File: Assets/Scripts/Spells/Straight.cs
using UnityEngine;
using System.Collections;

class Straight : MonoBehaviour
{
    public float Speed = 4F;
    public bool Moving = false;
    public bool Finished = false;
    public Vector3 Target;

    private float _incr = 0;

    void Update()
    {
        if (Finished)
            Destroy(gameObject, .01F);

        if (Moving)
        {
            if (_incr <= 1)
                _incr += Speed/100;
            else
                Finished = true;

            transform.position = Vector3.Lerp(transform.position, Target, _incr);
        }
    }
}
